code/game/machinery/computer/atmos_alert.dm

```diff
@@ -43,14 +43,22 @@
 	switch(action)
 		if("clear")
 			var/zone = params["zone"]
-			if(zone in priority_alarms)
-				to_chat(usr, "Priority alarm for [zone] cleared.")
-				priority_alarms -= zone
-				. = TRUE
-			if(zone in minor_alarms)
-				to_chat(usr, "Minor alarm for [zone] cleared.")
-				minor_alarms -= zone
-				. = TRUE
+			for(var/area/A in priority_alarms)
+				if(A.name == zone)
```

Comment:
this would need a little bit of work, ``zone`` should return an area rather than its name so we can search this, but it will be much better (or alternatively, ``priority_alarms`` can be a list of areas rather than area names)

```suggestion
			var/area/A = locate() in priority_alarms
			if(A)
```

Reply:
I tried this but tgui didn't work, because i don't know how to send the area type tgui